arches/management/commands/packages.py

```diff
@@ -1088,7 +1091,15 @@ def delete_tileserver_layer(self, layer_name=False):
                 tileserver_layer.map_layer.delete()
                 tileserver_layer.map_source.delete()
                 tileserver_layer.delete()
-
+    
+    def delete_mapbox_layer(self, layer_name=False):
+        if layer_name != False:
+            with transaction.atomic():
+                mapbox_layer = models.MapLayer.objects.get(name=layer_name)
+                for layerdef in mapbox_layer.layerdefinitions:
```

Comment:
@mradamcox since a single source may be used by many layers, it seems like you may find exceptions as a result of trying to delete sources that you have already deleted if you simply iterate over the layer definitions as you are doing here. 
I think we should instead create a set of unique source names, and then iterate over those and call delete on the models.  something like this might be a starting point (note: untested): 
```py
source_names = set([layerdef['source'] for layerdef in mapbox_layer.layerdefinitions])
for source_name in source_names:
    mapbox_layer_source = models.MapSource.objects.get(name=source_name)
    mapbox_layer_source.delete()
```

Reply:
Great catch on this, thanks. I'm finding that some layer defs don't actually have a `source` key, only `[u'paint', u'minzoom', u'type', u'id', u'layout']`. Thoughts on that? I assume those can just be skipped because they don't reference anything in the database.

Reply:
Also, I see that the sources are suffixed with the name of the layer, `[u'contours-Topo', u'hillshading-Topo', u'openmaptiles-Topo']` for example, where Topo is the name of the mapbox layer. This suggests that two different mapbox layers will never share the same source. Is that correct?